### PR TITLE
Document options in assets

### LINKED ISSUE #3950: Document webpack change: Grouping of assets and modules in stats for readablility
*A pull request by @sokra was merged and maintainers requested a documentation change.*

See pull request: https://github.com/webpack/webpack/pull/11409

---

**What kind of change does this PR introduce?**
feature

**Did you add tests for your changes?**
existing tests updates

**Does this PR introduce a breaking change?**
yes

**What needs to be documented once your changes are merged?**
New options:
* `stats.assetsSpace` how many items of assets should be displayed (groups will be collapsed to fit this space)
* `stats.modulesSpace` how many items of modules should be displayed (groups will be collapsed to fit this space)
* `stats.chunkModulesSpace` how many items of chunk modules should be displayed (groups will be collapsed to fit this space)
* `stats.nestedModulesSpace` how many items of nested modules should be displayed (groups will be collapsed to fit this space)
* `stats.cachedModules` (better named variant of `stats.cached`) should cached modules be displayed
* `stats.runtimeModules` (better named variant of `stats.runtime`) should runtime modules be displayed
* `stats.dependentModules` should chunk modules that are dependencies of other modules of the chunk be displayed (only for `stats.chunkModules`)
* `stats.groupAssetsByChunk` assets are grouped by chunk names
* `stats.groupAssetsByPath` and `stats.groupAssetsByExtension` assets are groups by path and/or extension
* `stats.groupAssetsByInfo` Group assets by their asset info (immutable, development, hotModuleReplacement, etc).
* `stats.groupAssetsByEmitStatus` Group assets by their emit status (emitted, compared for emit or cached).
* `stats.groupModulesByAttributes` Group modules by their attributes (errors, warnings, assets, optional, orphan, or dependent).
* `stats.groupModulesByCacheStatus` Group modules by their status (cached or built and cacheable).
* `stats.groupModulesByPath` and `stats.groupModulesByExtension` modules are groups by path and/or extension of the resource

Changed defaults:
* `stats.preset: "minimal"` shows number of assets
* for the text format all group options are enabled and assetsSpace: 15, modulesSpace: 15, nested/chunkModulesSpace: 10
* cachedModules defaults to false in the text format

Removed options:
* `stats.chunkRootModules` removed
* `stats.maxModules` removed
* `stats.runtime` removed

### PR DESCRIPTION
**Pull request by jeffin143**
fix #3950 

- [x] Read and sign the [CLA][1]. PRs that haven't signed it won't be accepted.
- [x] Make sure your PR complies with the [writer's guide][2].
- [x] Review the diff carefully as sometimes this can reveal issues.
- [x] Do not abandon your Pull Request: [Stale Pull Requests][3].

### CODE REVIEW

**Comment by EugeneHlushko** [comment]
This only removes the options, are you going to commit the new ones as well?

**Comment by jeffin143** [comment]
> This only removes the options, are you going to commit the new ones as well?

WIP

File: src/content/configuration/stats.md [hunk 1]
```diff
@@ -144,6 +144,51 @@ module.exports = {
 };
 ```
 
+### `stats.assetsSpace`
+
+`number`
+
+Tells `stats`  how many items of assets should be displayed (groups will be collapsed to fit this space)  .
```
Comment (montogeek) [inline src/content/configuration/stats.md:151]:
```suggestion
Tells `stats`  how many items of assets should be displayed (groups will be collapsed to fit this space).
```

File: src/content/configuration/stats.md [hunk 2]
```diff
@@ -144,6 +144,51 @@ module.exports = {
...
+    assetsSpace: 15
+  }
+};
+```
+
+### `stats.modulesSpace`
+
+`number`
+
+Tells `stats`  how many items of modules should be displayed(groups will be collapsed to fit this space)  .
```
Comment (montogeek) [inline src/content/configuration/stats.md:166]:
```suggestion
Tells `stats`  how many items of modules should be displayed(groups will be collapsed to fit this space).
```

File: src/content/configuration/stats.md [hunk 3]
```diff
@@ -144,6 +144,51 @@ module.exports = {
...
+    modulesSpace: 15
+  }
+};
+```
+
+### `stats.chunkModulesSpace`
+
+`number`
+
+Tells `stats`  how many items of chunk modules should be displayed(groups will be collapsed to fit this space)  .
```
Comment (montogeek) [inline src/content/configuration/stats.md:181]:
```suggestion
Tells `stats`  how many items of chunk modules should be displayed(groups will be collapsed to fit this space).
```

**Review by sokra — APPROVED** [review]

File: src/content/configuration/stats.md [hunk 4]
```diff
@@ -145,6 +145,171 @@ module.exports = {
 };
 ```
 
+### `stats.assetsSpace`
+
+`number = 15`
+
+Tells `stats`  how many items of assets should be displayed (groups will be collapsed to fit this space).
```
Comment (sokra) [inline src/content/configuration/stats.md:152]:
```suggestion
Tells `stats` how many items of assets should be displayed (groups will be collapsed to fit this space).
```

**Review by sokra — APPROVED** [review]

**Pull request merged**